### PR TITLE
feat: added filter functionality to filter out stackexchange questions/answers from the response source documents

### PR DESCRIPTION
**Pull request by Anyitechs**
feat: added filter functionality to filter out stackexchange questions/answers from the response source documents

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -0,0 +1,4643 @@
+lockfileVersion: '6.0'
```
Comment (adamjonas) [inline pnpm-lock.yaml]: do we need to check this in?

**Review by adamjonas — COMMENTED** [review]
all that are being returned are stackexchange answers. I wasn't able to generate any answers that retrieved other sources.

**Comment by Anyitechs** [comment]
> all that are being returned are stackexchange answers. I wasn't able to generate any answers that retrieved other sources.

I think this depends on the prompt. For example, questions on bips, like "what are bips?" will return answers from other sources, but a question on segwit, like "what is segwit?" will return answers from Stackexchange alone. I noticed this before the feature, so I think it depends on the question.

File: pnpm-lock.yaml (hunk 1 above)
Comment (Anyitechs) [reply pnpm-lock.yaml]: No, we don't have to. I just moved it to .gitignore file.

**Pull request merged**